Assets/Mirror/Runtime/NetworkReader.cs

```diff
@@ -10,51 +12,172 @@ namespace Mirror
     // The exceptions will be handled in NetworkServer/NetworkClient.
     public class NetworkReader
     {
-        // cache encoding instead of creating it with BinaryWriter each time
+        // internal buffer
+        byte[] buffer;
```

Comment:
Make this an ArraySegment<byte>

That way we read directly from the passed array instead of allocating a new one and copying the data.

Reply:
we already do.
this.buffer = buffer in the constructor should only copy the pointer.
but will probably make this an ArraySegment anyway for proper ArraySegment support in the constructor